### PR TITLE
November 2016 SIP meeting minutes

### PR DESCRIPTION
**Pull request by travissarles**

### CODE REVIEW

File: sips/minutes/_posts/2016-11-29-sip-minutes.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+Sébastien says binary compatibility is also an argument in favor of having explicit @static annotation. If in one version you have a static method, in the next version you add a method with same name and signature. The static implementation is not there anymore, then you have broken binary compatibility in a silent way.
+
+**Conclusion** There are a lot of edge cases when not using annotations. We need to work on this and think about cases around initialization to simplify it. Why do statics need to go first? Show surprising results. There are different implementation strategies. Is this more like @tailrec or does it change generated code?
+
+### [SIP-27: Trailing commas](http://docs.scala-lang.org/sips/completed/trailing-commas.html)
+
+Dale talks about how we wanted trailing comments for multi-line elements. Should be easy. Need to discuss which parts of the syntax can use trailing comments. There are two variants of the SIP to vote on. The first is _parameters and arguments_. The other variant is _everywhere_ for consistency. Dale implemented the first one. The second one shouldn't be more hard to implement except for tuples. It needs to fail compilation somehow.
```
Comment (Duhemm) [inline sips/minutes/_posts/2016-11-29-sip-minutes.md]: ~comments~ commas

File: sips/minutes/_posts/2016-11-29-sip-minutes.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+It could be confusing if trailing comments are only allowed some places and not others. We'd need lots of error messages.
```
Comment (Duhemm) [inline sips/minutes/_posts/2016-11-29-sip-minutes.md]: ~comments~ commas

File: sips/minutes/_posts/2016-11-29-sip-minutes.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+## Date and Location
+The meeting took place on 29 November 2016 via Google Hangouts at EPFL in Lausanne, Switzerland as well as other locations.
+
+Minutes were taken by Travis Lee.
+
+## Attendees
+
+* Martin Odersky ([@odersky](https://github.com/odersky)), EPFL
+* Jorge Vicente Cantero ([@jvican](https://github.com/jvican)), Process Lead
+* Seth Tisue ([@SethTisue](https://github.com/SethTisue)), EPFL
```
Comment (SethTisue) [inline sips/minutes/_posts/2016-11-29-sip-minutes.md]: Lightbend

**Comment by travissarles** [comment]
@jvican any other comments before we merge this?

File: sips/minutes/_posts/2016-11-29-sip-minutes.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+* Eugene Burmako ([@xeno-by](https://github.com/xeno-by)), EPFL
+* Iulian Dragos ([@dragos](https://github.com/dragos)), Independent
+* Adriaan Moors ([@adriaanm](https://github.com/adriaanm)), Lightbend
+* Dale Wijnand ([@dwijnand](https://github.com/dwijnand)), author of SIP-27
+
+
+## Proceedings
+### Opening Remarks
+
+**Jorge** We'll talk about the SIPS for META. Eugene will start.
```
Comment (jvican) [inline sips/minutes/_posts/2016-11-29-sip-minutes.md]: s/META/Scala Meta/g

File: sips/minutes/_posts/2016-11-29-sip-minutes.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+### [SIP-28 and SIP-29 - Inline meta](http://docs.scala-lang.org/sips/pending/inline-meta.html)
```
Comment (jvican) [inline sips/minutes/_posts/2016-11-29-sip-minutes.md]: s/Inline meta/Inline and Meta/g

File: sips/minutes/_posts/2016-11-29-sip-minutes.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+Eugene and co have been working hard for two months on inline and META. Previously discussed new macro system with new inline and meta features. Inline provides a facility to declare methods with inline right hand side into call side (0:01:24) and meta implements compile-time function execution to do meta-programming. Martin implemented inline mechanism in Dotty. Eugene worked on macro notations. New style macros will integrate with tools. Eugene shows how it works in IntelliJ. For example, you can print the value of the parameters. Meta blocks supported by IntelliJ. So are quasi-quotes. You can also expand macros. Will greatly help debugability.
+
+Iulian says we should flesh out the Scala meta API.
+
+The spec needs to be updated based on Martin's Dotty implementation. We need to split the SIPs for the next meeting.
+
+**Conclusion** More development needed. Must answer questions on PR.
```
Comment (jvican) [inline sips/minutes/_posts/2016-11-29-sip-minutes.md]: This proposal needs at least another iteration to shape up and provide concrete implementation and specification details. This proposal is therefore under revision -- Eugene, the author, will gather and address more feedback and will resubmit the proposal to analysis when it's ready.

File: sips/minutes/_posts/2016-11-29-sip-minutes.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+Heather says the debate is about the semantics or translation rules. All arguments are evaluated each time the parameter is referenced in the method. This is implemented in Dotty. Should this be implemented in Scalac?
+
+Sébastien thinks that Java bridge should be forbidden in this case. It's annotation that we add, so we could just make it an error.
+
+The SIP needs more examples. Martin says it's to remove an annoying restriction; there's no reason we can't we mix repeated and by name parameters. Evaluating them all together is much simpler. There's another reason regarding case classes. Inline function should maintain call by value semantics. Pass parameters by name just to be clean.
+
+The main motivation is to prepare for inline. Inline won't work very well without this. Need to flesh this out in SIP.
+
+**Conclusion** Someone needs to flesh out the SIP and how it relates to inline.
```
Comment (jvican) [inline sips/minutes/_posts/2016-11-29-sip-minutes.md]: There is not an implementation for Scalac, but for Dotty. This proposal is on hold until the Committee decides the specifics of the Inline proposal and how it relates to it. After that, the author will resubmit the proposal for further analysis.

File: sips/minutes/_posts/2016-11-29-sip-minutes.md [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+### [SIP-NN:Static](https://github.com/scala/scala.github.com/pull/491/files)
```
Comment (jvican) [inline sips/minutes/_posts/2016-11-29-sip-minutes.md]: SIP-25: Static

File: sips/minutes/_posts/2016-11-29-sip-minutes.md [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+
+Martin says it's important that there are these restrictions, that they come first in order not to have surprises with initialization order.
+
+A lot of people are surprised about initialization order.
+
+Sometimes we need something to be static for Java interop.
+
+Sébastien says binary compatibility is also an argument in favor of having explicit @static annotation. If in one version you have a static method, in the next version you add a method with same name and signature. The static implementation is not there anymore, then you have broken binary compatibility in a silent way.
+
+**Conclusion** There are a lot of edge cases when not using annotations. We need to work on this and think about cases around initialization to simplify it. Why do statics need to go first? Show surprising results. There are different implementation strategies. Is this more like @tailrec or does it change generated code?
```
Comment (jvican) [inline sips/minutes/_posts/2016-11-29-sip-minutes.md]: s/we need to work on this/The authors need time to work on the specifics of the proposal and address the Committee's feedback: .../g

Comment (jvican) [inline sips/minutes/_posts/2016-11-29-sip-minutes.md]: Can we add "This is the first review iteration of this proposal." at the beginning of the conclusion?

**Pull request merged**